src/app-layer-dns-tcp.c

```diff
@@ -508,6 +508,8 @@ static int DNSReponseParseData(Flow *f, DNSState *dns_state, const uint8_t *inpu
             tx->recursion_desired = 1;
     }
 
+    tx->flags = ntohs(dns_header->flags);
```

Comment:
Scan-build 5.0 is unhappy here:
<pre>
app-layer-dns-tcp.c:511:15: warning: Access to field 'flags' results in a dereference of a null pointer (loaded from variable 'tx')
    tx->flags = ntohs(dns_header->flags);
    ~~        ^
1 warning generated.
</pre>
It looks like it's correct. Other code is explicitly checking before dereferencing tx.

Reply:
And sure enough, my fuzzer managed to trip it:
<pre>
ASAN:SIGSEGV
=================================================================
==22062==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000004 (pc 0x000000721403 bp 0x7f76267e76d0 sp 0x7f76267e70a0 T20)
    #0 0x721402 in DNSReponseParseData /home/victor/qa/buildbot/donkey/z600fuzz/Private/src/app-layer-dns-tcp.c:511:17
    #1 0x71d349 in DNSTCPResponseParse /home/victor/qa/buildbot/donkey/z600fuzz/Private/src/app-layer-dns-tcp.c:589:17
    #2 0x7c89c5 in AppLayerParserParse /home/victor/qa/buildbot/donkey/z600fuzz/Private/src/app-layer-parser.c:1041:13
    #3 0x53bcce in TCPProtoDetect /home/victor/qa/buildbot/donkey/z600fuzz/Private/src/app-layer.c:421:17
    #4 0x53880f in AppLayerHandleTCPData /home/victor/qa/buildbot/donkey/z600fuzz/Private/src/app-layer.c:577:13
    #5 0x123f8c0 in ReassembleUpdateAppLayer /home/victor/qa/buildbot/donkey/z600fuzz/Private/src/stream-tcp-reassemble.c:1033:13
    #6 0x123cd15 in StreamTcpReassembleAppLayer /home/victor/qa/buildbot/donkey/z600fuzz/Private/src/stream-tcp-reassemble.c:1106:12
    #7 0x124bd7a in StreamTcpReassembleHandleSegment /home/victor/qa/buildbot/donkey/z600fuzz/Private/src/stream-tcp-reassemble.c:1724:13
    #8 0x1176ff0 in StreamTcpPacket /home/victor/qa/buildbot/donkey/z600fuzz/Private/src/stream-tcp.c:4514:17
    #9 0x11e2bd9 in StreamTcp /home/victor/qa/buildbot/donkey/z600fuzz/Private/src/stream-tcp.c:4929:11
    #10 0xe6fc41 in FlowWorker /home/victor/qa/buildbot/donkey/z600fuzz/Private/src/flow-worker.c:215:9
    #11 0x12a12f4 in TmThreadsSlotVarRun /home/victor/qa/buildbot/donkey/z600fuzz/Private/src/tm-threads.c:128:17
    #12 0x12c39fb in TmThreadsSlotVar /home/victor/qa/buildbot/donkey/z600fuzz/Private/src/tm-threads.c:585:17
    #13 0x7f7639cf3183 in start_thread /build/eglibc-SvCtMH/eglibc-2.19/nptl/pthread_create.c:312
    #14 0x7f7637983ffc in clone /build/eglibc-SvCtMH/eglibc-2.19/misc/../sysdeps/unix/sysv/linux/x86_64/clone.S:111
</pre>

Reply:
yes correct. I am going to provide the fix for this. Thanks.